algebird-core/src/main/scala/com/twitter/algebird/MapAlgebra.scala

```diff
@@ -222,4 +224,31 @@ object MapAlgebra {
 
   def dot[K, V](left: Map[K, V], right: Map[K, V])(implicit mring: Ring[Map[K, V]], mon: Monoid[V]): V =
     Monoid.sum(mring.times(left, right).values)
+
+  def cube[K, V](it: TraversableOnce[(K, V)])(implicit c: Cuber[K]): Map[c.K, TraversableOnce[V]] = {
+    val sg = new Semigroup[Iterable[V]] {
+      def plus(x: Iterable[V], y: Iterable[V]) = x ++ y
+    }
+    cubeSum(it.map { case (k, v) => (k, Iterable(v)) })(c, sg)
```

Comment:
The trade-off here is to allocate lots more objects (the Semigroup, lots of Iterable singletons and Maps) in order to take advantage of sumByKey (rather than using Scala's groupBy). Is this a good idea?